api/src/main/resources/app.py

```diff
@@ -82,6 +82,10 @@ def finish():
                 logging.info(ex.msg)
                 self.set_status(403)
                 self.finish(ex.msg)
+            elif isinstance(ex, FailedPrecondition):
+                logging.info(ex.msg)
+                self.set_status(412)
```

Comment:
I don't think HTTP 412 is appropriate here, let's just go with 400 for client error.